mempool/cat/reactor.go

```diff
@@ -459,8 +479,18 @@ func (memR *Reactor) broadcastSeenTxWithHeight(txKey types.TxKey, height int64,
 		},
 	}
 
-	selectedPeers := selectStickyPeers(signer, memR.ids.GetAll(), maxSeenTxBroadcast, memR.currentStickyPeerSalt())
-	for _, peerInfo := range selectedPeers {
+	peers := memR.ids.GetAll()
+	if len(peers) == 0 {
+		return
+	}
+
+	orderedPeers := selectStickyPeers(signer, peers, len(peers), memR.currentStickyPeerSalt())
+	sent := 0
+	for _, peerInfo := range orderedPeers {
+		if sent >= maxSeenTxBroadcast {
+			break
+		}
```

Comment:
not strictly necessary, but this seems like a slightly better approach if queues get full